File: server/src/commands/CommandFactory.js
const { MoveLeftCommand } = require('./MoveLeftCommand');
const { MoveRightCommand } = require('./MoveRightCommand');
const { StopCommand } = require('./StopCommand');
const { JumpCommand } = require('./JumpCommand');
const { SpawnItemCommand } = require('./SpawnItemCommand');
const { AddScoreCommand } = require('./AddScoreCommand');
const { Tree1Command } = require('./Tree1Command');
const { Tree2Command } = require('./Tree2Command');

const _commands = {
  MOVE_LEFT: new MoveLeftCommand(),
  MOVE_RIGHT: new MoveRightCommand(),
  STOP: new StopCommand(),
  JUMP: new JumpCommand(),
};

function create(type, arg, flag, db, bounds) {
  const key = (type || '').toUpperCase();
  if (key === 'SPAWN_ITEM')
    return new SpawnItemCommand(arg, db, bounds);
  if (key === 'ADD_SCORE')
    return new AddScoreCommand(arg);
  if (key === 'TREE1_COMMAND'){
    console.log('reached TREE1COMMAND FACTORY js');
    return new Tree1Command(flag);
    }
  if (key === 'TREE2_COMMAND'){
    console.log('reached TREE2COMMAND FACTORY js');
    return new Tree2Command(flag);
  }

  return _commands[key] || null;
}

function getValidTypes() {
  return [...Object.keys(_commands), 'SPAWN_ITEM itemId x y (e.g. gem 0 0)', 'ADD_SCORE amount'];
}

module.exports = { create, getValidTypes };

File: socketrunner_unity/Assets/Scripts/Commands/CommandFactory.cs
using UnityEngine;

public static class CommandFactory
{
	public static ICommand Create(NetworkData data, PlayerController player, TowerCamera camera)
	{
		if (data == null)
			return null;

		string type = data.type?.ToUpperInvariant();
		if (type == "ADD_SCORE")
		{
			int amount = 100;
			if (data.payload != null && data.payload.amount > 0)
				amount = (int)data.payload.amount;
			return new AddScoreCommand(amount);
		}

		if (type == "TREE1_COMMAND")
		{
			Debug.Log("Reached CommandFactory TREE1");
			bool flag = true;
			if (data.payload != null && data.payload.flag == true)
				flag = (bool)data.payload.flag;
			return new Tree1Command(flag);

		}

		if (type == "TREE2_COMMAND")
		{
			Debug.Log("Reached CommandFactory TREE2");
			bool flag = true;
			if (data.payload != null && data.payload.flag == true)
				flag = (bool)data.payload.flag;
			return new Tree2Command(flag);

		}

		if (player == null)
			return null;

		switch (type)
		{
			case "MOVE_LEFT":   return new MoveLeftCommand(player);
			case "MOVE_RIGHT":  return new MoveRightCommand(player);
			case "STOP":        return new StopCommand(player);
			case "JUMP":        return new JumpCommand(player);
			case "SPAWN_ITEM":
				string prefabType = data.payload?.prefabType;
				float x = data.payload?.x ?? 0f;
				float y = data.payload?.y ?? 0f;
				return string.IsNullOrEmpty(prefabType) ? null : new SpawnItemCommand(player, prefabType, x, y);

			default:
				return null;
		}
	}
}
